deepspeed/runtime/zero/stage3.py

```diff
@@ -2833,3 +2833,88 @@ def _handle_overflow(cpu_sum, x, i):
         logger.info(
             f"rank {rank} detected overflow {cpu_sum} in tensor {i}:{t_i} shape {x.shape}"
         )
+
+
+def estimate_zero3_mem_needs(model,

```

Comment:
I think its ok that this stays in stage3, but how would you suggest to expose this on the client side? something like deepspeed.utils.estimate_zero3_memory(..) ?@tjruwase @jeffra for suggestion as well

Reply:
Wherever you guys think is the best I will put it there.

Reply:
It'd be good to rename the function for it to be more intuitive at what mem needs it measures.
What would be a good collective name for param+optim_state+grad, i.e. everything but activations.

Reply:
We have been using model states to mean param+optim_state+grad. Lets use that?

Reply:
That works! 

So something like `estimate_zero3_model_state_mem_needs`? (same for others?)

Reply:
sounds good to me